src/components/Login/Login.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState, useRef, useEffect } from "react";
+import './Login.scss';
+import { Input } from "../Input";
+import { SignForm } from '../SignForm/SignForm'
+import { NamePage } from "../NamePage/NamePage";
+
+export const Login = () => {
+    
+    const [valueEmail, setValueEmail] = useState('')
+    const [errorEmail, setErrorEmail] = useState('')
+    
+    const changeInputEmail = (event: any): void => {
+        
+        setValueEmail(event.target.value)
+        
+        event.target.onblur=function(){
+
+            const checkEmail = /[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/;
+
+            if(checkEmail.test(event.target.value) === false && event.target.value != '' ){
+                setErrorEmail('Invalid email')     
+            }; 
+        }
+        event.target.onfocus=function(){
+            setErrorEmail('')
+        }
```

Comment:
Лучше выносить такие функции в отдельные функции и навешивать их на сам элемент через ref в useEffect. Либо навешивать на сам элемент кнопки как обычный обработчик, как onClick (onBlur, onFocus)
```suggestion
useEffect(() => {
   someRef.current.addEventListener('blur', someBlurFunction))
   return () => {  someRef.current.removeEventListener('blur', someBlurFunction))
}, [])
       
```